src/core/thread/mle.cpp

```diff
@@ -1725,51 +1720,82 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     // Active Timestamp
     if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
     {
+        const MeshCoP::Timestamp *timestamp;
+
         VerifyOrExit(activeTimestamp.IsValid(), error = kThreadError_Parse);
+        timestamp = mNetif.GetActiveDataset().GetNetwork().GetTimestamp();
 
-        // Active Dataset
-        if (Tlv::GetOffset(aMessage, Tlv::kActiveDataset, offset) == kThreadError_None)
+        // if received timestamp does not match the local value and message does not contain the dataset,
+        // send MLE Data Request
+        if ((timestamp == NULL || timestamp->Compare(activeTimestamp) != 0) &&
+            (Tlv::GetOffset(aMessage, Tlv::kActiveDataset, activeDatasetOffset) != kThreadError_None))
         {
-            aMessage.Read(offset, sizeof(tlv), &tlv);
-            mNetif.GetActiveDataset().Set(activeTimestamp, aMessage, offset + sizeof(tlv), tlv.GetLength());
-        }
-        else
-        {
-            dataRequest = true;
+            ExitNow(dataRequest = true);
         }
     }
+    else
+    {
+        activeTimestamp.SetLength(0);
```

Comment:
If the Data Response has no Active Timestamp TLV, then we remember that by setting the `activeTimestamp` length to zero.  Further below where we actually perform the updates to the Network Data, Active Dataset, and Pending Datasets, if the length of `activeTimestamp` is zero, then we clear the copy of the network dataset.